ksy_schema.json

```diff
@@ -99,16 +99,23 @@
       "type": "object",
       "additionalProperties": false,
       "properties": {
-        "id": { "$ref": "#/definitions/Identifier" },
-        "title": { "type": "string" },
+        "id": {
+          "$ref": "#/definitions/Identifier",
+          "description": "unique string that identifies this format. It is recommended to use the same name, as basename of the file with the format description. This name also will be used to infer name of type when generate parser\n\nThis field is mandatory on top-level definition and shouldn't be used at nested levels"
```

Comment:
> It is recommended to use the same name, as basename of the file with the format description.

I'm wondering if we should say "It should have the same name [...]" instead of just "recommended". Are there any good use cases for having a spec whose `id` doesn't match the file name, and does the compiler even support this properly? I've never really tried it, so I don't know how well it works.

Minor thing: the term "basename" isn't correct here - at least according to the Unix shell command `basename`, the basename of a path still includes the file extension:

```sh
$ basename path/to/some.ksy
some.ksy
```

I would write it in plain English instead - something like "It should be identical to the file name without the `.ksy` extension".

> This name also will be used to infer name of type when generate parser

I would remove the "also" here - this is basically the main purpose of `id`, not a minor side effect.

Reply:
> Are there any good use cases for having a spec whose id doesn't match the file name, and does the compiler even support this properly?

I'm not sure, but I think I've see that somewhere in the Kaitai user manual... never-mind, replaced with your suggestion